File: client/src/app/modules/tiers/services/billing.label.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { BillingLabelType } from '../../tiers/model/BillingLabelType';

@Injectable({
  providedIn: 'root'
})
export class BillingLabelTypeService extends AppRestService<BillingLabelType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getBillingLabelTypes() {
    return this.getListCached(new HttpParams(), "billing-label-types");
  }

  addOrUpdateBillingLabelType(billingLabelType: BillingLabelType) {
    this.clearListCache(new HttpParams(), "billing-label-types");
    return this.addOrUpdate(new HttpParams(), "billing-label-type", billingLabelType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/BillingLabelTypeService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.modules.osiris.tiers.model.BillingLabelType;

public interface BillingLabelTypeService {
    public List<BillingLabelType> getBillingLabelTypes();

    public BillingLabelType getBillingLabelType(Integer id);

    public BillingLabelType addOrUpdateBillingLabelType(BillingLabelType billingLabelType);
}
